pkg/operatorhub/operatorhub.go

```diff
@@ -0,0 +1,65 @@
+package operatorhub
+
+import (
+	"sync"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/operator-framework/operator-marketplace/pkg/defaults"
+)
+
+// DefaultName is the default name of the OperatorHub config resource on an
+// OpenShift cluster
+const DefaultName = "cluster"
+
+// instance is the singleton hubconfig object
+var instance *operatorhub
+
+func init() {
+	instance = &operatorhub{
+		current: make(map[string]bool),
+		lock:    sync.Mutex{},
+	}
+}
+
+// operatorhub implements OperatorHub
+type operatorhub struct {
+	current map[string]bool
+	lock    sync.Mutex
+}
+
+// OperatorHub is the interface to interact with the OperatorHub configuration in
+//  memory.
+type OperatorHub interface {
+	Get() map[string]bool
+	Set(hub configv1.OperatorHubSpec)
+}
+
+// New returns the singleton instance of HubConfig
+func New() OperatorHub {
```

Comment:
Sure. I will go with `GetSingleton()` give `OperatorHub` is redundant as per go conventions. BTW, in `Defaults` it is actually instantiating a new instance, so that can stay as `New()`